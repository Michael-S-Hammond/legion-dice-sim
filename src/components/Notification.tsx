import '../css/Notification.css';

import React from 'react';

type NotificationProps = {
    message: string;
}

class Notification extends React.Component<NotificationProps> {
    constructor(props : NotificationProps) {
        super(props);
    }

    render() : JSX.Element {
        return (
            <div className="alert alert-secondary alert-dismissible fade show mb-0" role="alert">
                {this.props.message}
                <button type="button" className="close" data-dismiss="alert" aria-label="Close">
                    <span aria-hidden="true">&times;</span>
                </button>
            </div>
        );
    }
}

export default Notification;
