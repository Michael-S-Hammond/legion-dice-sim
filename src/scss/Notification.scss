.alert {
    border-radius: 0 !important;
    text-align: center;
}
