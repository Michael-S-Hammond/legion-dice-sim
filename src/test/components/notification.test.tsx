import { create } from 'react-test-renderer';

import Notification from '../../components/Notification';

describe('Notification', () => {
    it('matches the snapshot', () => {
        const snapshot = create(<Notification message='This is a test.'></Notification>)
        expect(snapshot.toJSON()).toMatchSnapshot();
    });
});
