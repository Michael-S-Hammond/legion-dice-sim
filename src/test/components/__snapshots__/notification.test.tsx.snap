// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification matches the snapshot 1`] = `
<div
  className="alert alert-secondary alert-dismissible fade show mb-0"
  role="alert"
>
  This is a test.
  <button
    aria-label="Close"
    className="close"
    data-dismiss="alert"
    type="button"
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
  </button>
</div>
`;
